src/app/[locale]/(main)/overview/resources/[categoryName]/[[...subcategoryName]]/page.tsx

```diff
@@ -0,0 +1,78 @@
+import { Category, Configuration } from '@/schemas/configurationSchema';
+import Overview, { OverviewRow } from '@/components/overview/Overview';
+import { concatenateNameAndId, replaceWhitespaceInString, safeStringCompare } from '@/utils/stringUtils';
+
+import ErrorComponent from '@/components/error-handling/ErrorComponent';
+import PageWrapper from '@/components/layout/PageWrapper';
+import { getIconForResource } from '@/utils/icons';
+import { getValidatedConfiguration } from '@/schemas/validate';
+
+export default async function Page({
+  params: { categoryName, subcategoryName },
+}: {
+  params: { categoryName: string; subcategoryName?: string[] };
+}) {
+  const { success, configuration, error } = await getValidatedConfiguration();
+  if (!success) {
+    return <ErrorComponent type="configurationError" error={error} />;
+  }
+
+  const category = configuration.categories.find((item) => safeStringCompare(categoryName, item.name));
+  if (!category) {
+    return <ErrorComponent type="notFound" />;
+  }
+
+  if (!subcategoryName || subcategoryName.length === 0) {
+    return (
+      <PageWrapper title={category.name} description={category.description}>
+        <Overview content={getCategoryContent(configuration, category)} />
+      </PageWrapper>
+    );
+  }
+
+  const [subcategoryNameString] = subcategoryName;
+  const subcategory = category.subcategories?.find((sub) => safeStringCompare(subcategoryNameString, sub.name));
+  if (subcategoryName.length > 1 || !subcategory) {
+    return <ErrorComponent type="notFound" />;
+  }
+
+  return (
+    <PageWrapper title={subcategory.name} description={subcategory.description}>
+      <Overview content={getCategoryContent(configuration, subcategory)} />
+    </PageWrapper>
+  );
+}
+
+function getCategoryContent(configuration: Configuration, category: Category) {
+  const subcategories =
+    category.subcategories?.map((subcategory) => ({
+      title: subcategory.name,
+      description: subcategory.description,
+      href: `/overview/resources/${replaceWhitespaceInString(category.name).toLowerCase()}/${replaceWhitespaceInString(subcategory.name).toLowerCase()}`,
+      isCategory: true,
+      icon: subcategory.icon,
+    })) ?? [];
+  const resources = getResourcesForCategory(configuration, category);
```

Comment:
If you would transfer the information whether or not a subcategory is present here, you could short-circuit and simply return resources. Was there a reason to no longer simply push everything onto the same array?

Reply:
I've rearranged this a bit in order the implement the suggested short-circuit.

Reason for not pushing into the same array was the ability to separate the logic a bit by separately mapping `resources` and `category.subcategories`.